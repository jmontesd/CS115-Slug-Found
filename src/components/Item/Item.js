import React from 'react';
import './Item.scss';
import { Link } from 'react-router-dom';
import moment from 'moment';

export const Item = (props) => {
  // get the post from props
  const { post } = props;
  // render this as html
  return (
    <div className="container" id="post">
      <div className="postText">
          <Link to={`/item/${post.id}`}>{post.title}</Link>
      </div>
      <div className=".text"> {`${moment(post.createdAt).calendar()}`} </div>
      <Link to={`/item/${post.id}`}>
        <img alt="item" src={post.imageURL} />
      </Link>
      <button className="button" id="deleteBtn"> Delete </button> 
      {/* do only for personal posts */}
      <div className="divider"/>
      {/* <button className="button" id="messageBtn"> Message </button> */ /*add only for item page*/}
      
    </div>
  );
};

export default Item; 
