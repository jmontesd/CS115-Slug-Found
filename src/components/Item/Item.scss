.button {
  background-color: rgba(0, 153, 255, 0.753); /* Green */
  border-radius: 4px;
  border: none;
  border-color: white;
  color: whitesmoke;
  padding: 5px 5px;
  font-size: 14px;
  border-collapse: separate;
  position: absolute;
  border-radius: 25px;
}

.item-image {
  height: 85%;
  width: 55%;
  float: right;
  top: 18px;
  right: 8px;
  position: absolute;
}

.divider {
  width: 5px;
  height: auto;
  display: inline-block;
}

.text {
  font-family: 'Courier New', Courier, monospace;
}

#deleteBtn {
  top: 205px;
  left: 30px;
}

#post {
  margin: 0 auto;
  margin-bottom: 10px;
  background-color: white;
  border-radius: 25px;
  border: 4px solid clear;
  box-shadow: 5px 8px 7px #888888;
  border-color: white;
  border-spacing: 15px 200px;
  border-collapse: separate;
  padding: 20px;
  width: 550px;
  height: 250px;
  display: block;
  position: relative;
}
