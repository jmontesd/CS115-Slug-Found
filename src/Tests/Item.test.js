import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Item } from '../components/Item/Item';

// needed for shallow rendering
configure({ adapter: new Adapter() });

describe('<Item />', () => {
  it('should render Item properly', () => {
    const wrapper = shallow(
      <Item
        post={{ id: '1', createAt: 0, title: 'title', imageURL: 'http://placehold.it/300x300' }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
