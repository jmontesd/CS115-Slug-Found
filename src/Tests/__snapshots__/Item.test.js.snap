// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render Item properly 1`] = `ShallowWrapper {}`;
